### DISCUSSION TITLE
What to display when printing?

### DISCUSSION THREAD

**Discussion by ciar4n** [joomla/joomla-cms#47194] [2. Cold Features]
A general question for both 3.8 and 4.0 frontend template... what should be displayed when printing a page via the browser? Currently everything which doesn't really make for a very readable page. 

1. Hide every link been appended with a URL? Wikipedia for example hides this.
2. Should all modules be displayed? It is possible to show/hide with a module class.
3. Remove layout styling? Eg borders
4. Just display the page title and component area?
5. A pointless question considering there is an action for this (./templates/protostar/component.php)?

Current...

![image](https://user-images.githubusercontent.com/2803503/33097142-1ed21568-cf01-11e7-9f35-0d0d0c394f43.png)

The 4.0 template currently has some larger print issues.

**Comment by dgrammatiko** [comment]
I have no clue about the content part, my only suggestion/proposal is to **not** load this mostly unneeded css file in the head (optimally it should be loaded async after page complete, eg through js with a fallback inside a noscript tag)

**Comment by ghost** [comment]
By Core should be displayed

- Site-Name
- URL of showen Page
- Content (Article, Blog- or List-View,…)

Feature Request: Default Sets Super-Users can choose (Raw one like above to Full print all on Page).
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18789">issues.joomla.org/tracker/joomla-cms/18789</a>.</sub>

**Comment by coolcat-creations** [comment]
* Sitename and URL of the current page
* Website Logo
* Page Title
* Content Title (Article)
* Content
* Modules on/off

**Comment by Hackwar** [comment]
Joomla 3.x has a feature freeze and Joomla 4.0 has removed the specific printing button. Defining a print view with that falls under the duty of the template designer/site integrator. Thus this issue is not one of the Joomla project anymore. I would like to ask to close this one.

[a comment by an automated account is not shown]

**Comment by brianteeman** [comment]
Please re-open. We need a print.css for the frontend template in joomla 4 - thats what this is about

[a comment by an automated account is not shown]

**Comment by wilsonge** [comment]
Agree with Brian here. Actually with removing the print thing we should have a print.css that probably removes all modules and just prints the main content of the page (i.e. a version of the `component.php` layout).

**Comment by brianteeman** [comment]
Is it too late to do this?

**Comment by brianteeman** [comment]
Could someone please change the tag on this from J3 to J4

**Comment by Mohammadkhan18** [comment]
This are things should be done for display when printing 
1:Hide unnecessary elements
2:Remove unnecessary styling
3:Display only relevant content
4:Provide a print-friendly layout: Creating a separate stylesheet for printing can help to optimize the layout and styling for printed pages.

**Comment by Mohammadkhan18** [comment]
i want to contribute on this issue but i am confused after cloning ,i have open the template file and again in the template file there is two files "autm" and "system" .so where should i made my print.css

**Comment by drmenzelit** [comment]
@Mohammadkhan18 sorry for taking so long to answer. We are here talking about the frontend template Cassiopeia. On a clon of Joomla you will find the assets in the build folder: https://github.com/joomla/joomla-cms/tree/4.4-dev/build/media_source/templates/site/cassiopeia/scss
